From: Ruslan Ermilov <ru@nginx.com>
Date: Thu, 24 May 2018 12:06:35 +0300
Subject: Allowed digits, '+', '-', and '.' in scheme names as per RFC 3986.

--- a/src/http/ngx_http_parse.c
+++ b/src/http/ngx_http_parse.c
@@ -307,6 +307,11 @@ ngx_http_parse_request_line(ngx_http_req
                 break;
             }
 
+            if ((ch >= '0' && ch <= '9') || ch == '+' || ch == '-' || ch == '.')
+            {
+                break;
+            }
+
             switch (ch) {
             case ':':
                 r->schema_end = p;
