From: Maxim Dounin <mdounin@mdounin.ru>
Date: Wed, 13 Sep 2017 15:51:58 +0300
Subject: Fixed reference to time parsing code after 8b6fa4842133.

--- a/src/core/ngx_times.c
+++ b/src/core/ngx_times.c
@@ -320,7 +320,7 @@ ngx_gmtime(time_t t, ngx_tm_t *tp)
 
     /*
      * the algorithm based on Gauss' formula,
-     * see src/http/ngx_http_parse_time.c
+     * see src/core/ngx_parse_time.c
      */
 
     /* days since March 1, 1 BC */
