From: Maxim Dounin <mdounin@mdounin.ru>
Date: Thu, 7 Dec 2017 17:09:36 +0300
Subject: Configure: fixed SO_BINDANY comment.

--- a/auto/unix
+++ b/auto/unix
@@ -342,7 +342,7 @@ ngx_feature_test="setsockopt(0, SOL_SOCK
 . auto/feature
 
 
-# NetBSD bind to any address for transparent proxying
+# OpenBSD bind to any address for transparent proxying
 
 ngx_feature="SO_BINDANY"
 ngx_feature_name="NGX_HAVE_TRANSPARENT_PROXY"
