From: Debayan Ghosh <debayang.qdt@qualcommdatacenter.com>
Date: Mon, 11 Dec 2017 15:05:35 +0000
Subject: Configure: set default cacheline size to 64 for aarch64 platforms.

--- a/auto/os/conf
+++ b/auto/os/conf
@@ -110,6 +110,11 @@ case "$NGX_MACHINE" in
         NGX_MACH_CACHE_LINE=64
     ;;
 
+    aarch64 )
+        have=NGX_ALIGNMENT value=16 . auto/define
+        NGX_MACH_CACHE_LINE=64
+    ;;
+
     *)
         have=NGX_ALIGNMENT value=16 . auto/define
         NGX_MACH_CACHE_LINE=32
